lib/Syntax/SyntaxData.cpp

```diff
@@ -73,27 +73,35 @@ RC<SyntaxData> SyntaxData::getNextNode() const {
   if (hasParent()) {
     for (size_t I = getIndexInParent() + 1, N = Parent->getNumChildren();
          I != N; I++) {
-      if (auto C = getParent()->getChild(I))
-        return C;
+      if (auto C = getParent()->getChild(I)) {
+        if (C->getRaw()->isPresent() && C->getFirstToken())
```

Comment:
If we skip over missing nodes here, I think we should do the same in `getPreviousNode`. Also I think it's worth documenting that `getNextNode` will return the next node that does contain a non-missing token.